### ISSUE TITLE
kitchen verify still tries to install busser regardless of being already installed in kitchen converge

### ISSUE THREAD

**Issue by atifrasheed79** [test-kitchen/test-kitchen#1514]
## Description

My kitchen test was failing as busser-serverspec was trying to install latest version of bundler which requires ruby 2.3, so I installed bundler v1.17.3, busser, busser-serverspec via bootstrap.sh but "kitchen verify" still tries to install busser regardless of being already installed during "kitchen converge". 

## Kitchen Version
Test Kitchen version 1.16.0

## Ruby Version
ruby 2.0.0p648 (2015-12-16) [x86_64-linux]

## Platform Version
Amazon Linux release 2 (Karoo)

## Kitchen Output
$ bundle exec kitchen test
-----> Starting Kitchen (v1.16.0)
-----> Cleaning up any prior instances of <default-codeninjas-rhel-7-java-18>
-----> Destroying <default-codeninjas-rhel-7-java-18>...
       Finished destroying <default-codeninjas-rhel-7-java-18> (0m0.00s).
-----> Testing <default-codeninjas-rhel-7-java-18>
-----> Creating <default-codeninjas-rhel-7-java-18>...
       If you are not using an account that qualifies under the AWS
free-tier, you may be charged to run these suites. The charge
should be minimal, but neither Test Kitchen nor its maintainers
are responsible for your incurred costs.

       Instance <i-05e5918e9997e181e> requested.
       Polling AWS for existence, attempt 0...
       Attempting to tag the instance, 0 retries
       EC2 instance <i-05e5918e9997e181e> created.
       Waited 0/300s for instance <i-05e5918e9997e181e> volumes to be ready.
       Waited 5/300s for instance <i-05e5918e9997e181e> volumes to be ready.
       Waited 0/300s for instance <i-05e5918e9997e181e> to become ready.
       Waited 5/300s for instance <i-05e5918e9997e181e> to become ready.
       Waited 10/300s for instance <i-05e5918e9997e181e> to become ready.
       Waited 15/300s for instance <i-05e5918e9997e181e> to become ready.
       Waited 20/300s for instance <i-05e5918e9997e181e> to become ready.
       Waited 25/300s for instance <i-05e5918e9997e181e> to become ready.
       Waited 30/300s for instance <i-05e5918e9997e181e> to become ready.
       EC2 instance <i-05e5918e9997e181e> ready.
:paranoid is deprecated, please use :verify_host_key. Supported values are exactly the same, only the name of the option has changed.
       [SSH] Established
       Finished creating <default-codeninjas-rhel-7-java-18> (0m47.56s).
-----> Converging <default-codeninjas-rhel-7-java-18>...
       Preparing files for transfer
       Preparing script
       Transferring files to <default-codeninjas-rhel-7-java-18>
Fetching: bundler-1.17.3.gem (100%)
       Successfully installed bundler-1.17.3
       Parsing documentation for bundler-1.17.3
       Installing ri documentation for bundler-1.17.3
       1 gem installed
Fetching: thor-0.19.0.gem (100%)
       Successfully installed thor-0.19.0
Fetching: busser-0.7.1.gem (100%)
       Successfully installed busser-0.7.1
       Parsing documentation for thor-0.19.0
       Installing ri documentation for thor-0.19.0
       Parsing documentation for busser-0.7.1
       Installing ri documentation for busser-0.7.1
       2 gems installed
Fetching: rake-12.3.2.gem (100%)
       Successfully installed rake-12.3.2
Fetching: rspec-support-3.8.0.gem (100%)
       Successfully installed rspec-support-3.8.0
Fetching: rspec-core-3.8.0.gem (100%)
       Successfully installed rspec-core-3.8.0
Fetching: busser-serverspec-0.5.10.gem (100%)
       Successfully installed busser-serverspec-0.5.10
       Parsing documentation for rake-12.3.2
       Installing ri documentation for rake-12.3.2
       Parsing documentation for rspec-support-3.8.0
       Installing ri documentation for rspec-support-3.8.0
       Parsing documentation for rspec-core-3.8.0
       Installing ri documentation for rspec-core-3.8.0
       Parsing documentation for busser-serverspec-0.5.10
       Installing ri documentation for busser-serverspec-0.5.10
       4 gems installed
       Finished converging <default-codeninjas-rhel-7-java-18> (0m23.56s).
-----> Setting up <default-codeninjas-rhel-7-java-18>...
       Finished setting up <default-codeninjas-rhel-7-java-18> (0m0.00s).
-----> Verifying <default-codeninjas-rhel-7-java-18>...
       Preparing files for transfer
-----> Installing Busser (busser)
Fetching: thor-0.19.0.gem (100%)
       Successfully installed thor-0.19.0
Fetching: busser-0.7.1.gem (100%)
       Successfully installed busser-0.7.1
       2 gems installed
       Installing Busser plugins: busser-serverspec
       Plugin serverspec installed (version 0.5.10)
-----> Running postinstall for serverspec plugin
       /usr/share/rubygems/rubygems/installer.rb:518:in `ensure_required_ruby_version_met': bundler requires Ruby version >= 2.3.0. (Gem::InstallError)
        from /usr/share/rubygems/rubygems/installer.rb:781:in `pre_install_checks'
        from /usr/share/rubygems/rubygems/installer.rb:216:in `install'
        from /usr/share/rubygems/rubygems/dependency_installer.rb:374:in `block in install'
........
......

**Comment by atifrasheed79** [comment]
Just to update I was able to work it around by setting the BUSSER_HOME, GEM_HOME, GEM_PATH etc. in UserData (kitchen.yml) for the EC2 instance Kitchen will launch, and bootstrap.sh. 

When I enabled debug, I found that Kitchen converge and setup are working in a different environment when I was doing "Kitchen test", so I just made sure both (/all) steps refer to same GEM environment. 

So just enable debug when you are doing "Kitchen Test" to make sure all Kitchen Test phases are working in same GEM environment.

kitchen test --log-level=debug
